utils/utils.go

```diff
@@ -14,101 +14,238 @@ limitations under the License.
 package utils
 
 import (
+	"archive/zip"
+	"io"
 	"io/ioutil"
+	"log"
 	"net/http"
+	"os"
+	"path/filepath"
 
 	pkgerrors "github.com/pkg/errors"
 	appsV1 "k8s.io/api/apps/v1"
 	coreV1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
-// Download the raw CSAR file from a specific URL
-var Download = func(url string) ([]byte, error) {
+var CSAR FileOperator
+
+// FileOperator is an interface to Download, Extract and Delete files
+type FileOperator interface {
+	Download(string, string) error
+	Unzip(string) error
+	Delete(string) error
+}
+
+// CSARFile is a concrete type to Download, Extract and Delete CSAR files
+type CSARFile struct{}
+
+// Download a CSAR file
+func (c *CSARFile) Download(fileName string, url string) error {
+	file, err := os.Create(fileName)
```

Comment:
Shouldn't be nice to have a regex validator for the URL param?

Reply:
Well, as of now, since we don't know how the SDC url will look, I am keeping it as a placeholder.